sled-agent/src/services.rs

```diff
@@ -3201,7 +3212,7 @@ mod test {
             SidecarRevision::Physical("rev-test".to_string()),
             SWITCH_ZONE_BOOTSTRAP_IP,
             vec![],
-            StorageManager::new(&log, storage_key_requester).await,
+            StorageResources::new_for_test(),
```

Comment:
We don't need access to the full `StorageManager` here -- we aren't going to be adding new disks here -- we just want to know about "what U.2s exist on the system", which is a smaller interface.